overlord/snapstate/storehelpers.go

```diff
@@ -323,6 +324,29 @@ func updateInfo(st *state.State, snapst *SnapState, opts *RevisionOptions, userI
 		Flags:     storeFlags,
 	}
 
+	enforcedSets, err := EnforcedValidationSets(st)
+	if err != nil {
+		return nil, err
+	}
+	if enforcedSets != nil {
+		requiredValsets, requiredRevision, err := enforcedSets.CheckPresenceRequired(naming.Snap(curInfo.InstanceName()))
+		if err != nil {
+			return nil, err
+		}
+		if !requiredRevision.Unset() && snapst.Current == requiredRevision {
+			return nil, fmt.Errorf("snap %q is at the required revision %s by validation sets: %s, cannot update",
```

Comment:
I don't think we want a special new error here, we just want to reproduce the behavior if we got back the the current revision from the store which is handled here I think:

https://github.com/snapcore/snapd/blob/master/store/store_action.go#L612